dart/gui/osg/Viewer.cpp

```diff
@@ -843,6 +843,53 @@ const ::osg::ref_ptr<::osg::Group>& Viewer::getRootGroup() const
   return mRootGroup;
 }
 
+//==============================================================================
+void Viewer::setVerticalFieldOfView(double fov)
```

Comment:
Nitpick: `const double`

Reply:
Hm, I see the reason for this suggestion; I found this [SO post](https://stackoverflow.com/a/16033837/3122234) says specifying `const` modifier helps the compiler with optimizations. However, AFAIK, we haven't used `const` when we pass value parameters in DART codebase. Do you suggest we follow this convention from now on?

Reply:
I recommend using `const` for every variable which isn't intentionally mutable. Besides compiler optimizations, it also catches bugs where a variable might get unintentionally modified.

Note that you don't need the `const` in the declaration of the function. E.g.:

```
// in foo.hpp
void foo(int value);

// in foo.cpp
void foo(const int value)
{
  /* ... do things ... */
}
```

works perfectly fine, because the function signature is the same, whether or not a value parameter is const-qualified. On the other hand, function signatures are different between const-references and mutable lvalue references, so this would not work:

```
// in foo.hpp
void foo(int& value);

// in foo.cpp
void foo(const int& value)
{
  /* ... do things ... */
}
```

Reply:
To illustrate a potential bug, we might accidentally type

```
const bool result = camera->getProjectionMatrixAsPerspective(
      fov, aspectRatio, zNear, zFar);
```

instead of

```

const bool result = camera->getProjectionMatrixAsPerspective(
      fovy, aspectRatio, zNear, zFar);
```

which would accidentally overwrite the value that the caller passed to the function.

Reply:
Thanks for the examples. I see the point of preventing a potential bug, but I have a little concern with that their function signatures are not "exactly" the same. Wouldn't it be more clear to just put `const` to both of declaration and definition in this case?

Reply:
Just to be clear, the function signatures, in the technical sense, *are* **exactly** the same. A human might see that the text is different between the declaration and the definition, but the compiler/linker does not care, and the ABI signature will be identical.

There's nothing inherently wrong with putting the `const` in the declaration as well, although it's often discouraged because it's seen as unnecessary noise which a user has no reason to care about (it makes no difference whatsoever to a user whether a value parameter of a function is const-qualified or not, because it has no impact on how they use the function or how the function behaves).